src/Http/Controllers/Admin/MenuController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace CubeAgency\ArboryMenu\Http\Controllers\Admin;
+
+use Arbory\Base\Nodes\Node;
+use Arbory\Base\Admin\Form;
+use Arbory\Base\Admin\Form\FieldSet;
+use Arbory\Base\Admin\Grid;
+use CubeAgency\ArboryMenu\Menu\Menu;
+use Illuminate\Routing\Controller;
+use Arbory\Base\Admin\Traits\Crudify;
+
+class MenuController extends Controller
+{
+    use Crudify;
+
+    /**
+     * @var string
+     */
+    protected $resource = Menu::class;
+
+    /**
+     * @param Form $form
+     * @return Form
+     */
+    protected function form(Form $form)
+    {
+        $form->setFields(function (FieldSet $fields) {
+            $fields->text('name')->rules('required');
+            $fields->hasMany('items', function (FieldSet $fields) {
+                $fields->objectRelation('node', Node::class, 1);
+                $fields->text('name')->rules('required_without:resource.items.*.node.related_id');
+                $fields->text('link')->rules('nullable|url|required_without:resource.items.*.node.related_id')
+                    ->setLabel('link (replaces the node field value)');
+            })->sortable('position');
+        });
+
+        return $form;
+    }
+
+    /**
+     * @param Grid $grid
+     */
+    public function grid(Grid $grid)
+    {
+        $grid->column('name');
```

Comment:
Maybe add the related node?

Reply:
This is a god idea. At the moment menu module is not providing relation to node or specific object, it is done manually adding menu_id to node or specific objects. 
Do you mean finding node where this is related to or do you mean adding node relation to menu directly that can ease the implementation as well and there will not be a need to create a separate migrations?

Reply:
Nevermind, I mixed up menu_items with menu. Closed.